backend/src/controllers/PermissionController.ts

```diff
@@ -0,0 +1,26 @@
+import asyncHandler from 'express-async-handler';
+import { Request, Response } from 'express';
+import Permission from '../types/Perm';
+
+/**
+ * @desc 	Get all the roles
+ * @route 	GET /api/permissions/
+ */
+const getAllPermissions = asyncHandler(async (req: Request, res: Response) => {
+    const permissions = [];
+    for (const enumMember in Permission) {
+        const isNumber = parseInt(enumMember) >= 0;
+        if (!isNumber) {
+            permissions.push(enumMember);
+        }
```

Comment:
I've not tried to do this with enums in Typescript before, so I may be way off base here, but rather than trying to parse the int, perhaps try: `isNaN(enumMember)`.

Reply:
Doing that one gives the error:
Argument of type 'string' is not assignable to parameter of type 'number'

Not sure if there is a better work-around for the typing than the one I'm currently using but Typescript is too smart for NaN

Reply:
Ah, that makes sense because apparently, the numbers are strings too. 

Do you think that perhaps `isNaN(parseInt(enumMember))` is clearer?  I don't really mind though, I'm happy to stick with this :)